sdk/sdk/OBAApplication.h

```diff
@@ -0,0 +1,44 @@
+//
+//  sdk.h
+//  sdk
+//
+//  Created by Dima Belov on 4/25/15.
+//  Copyright (c) 2015 One Bus Away. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "OBAReferencesV2.h"
+#import "OBAModelDAO.h"
+#import "OBAModelService.h"
+#import "OBALocationManager.h"
+
+@interface OBAApplication : NSObject
+
+@property(nonatomic, strong, readonly) OBAReferencesV2 * references;
+@property(nonatomic, strong, readonly) OBAModelDAO * modelDao;
+@property(nonatomic, strong, readonly) OBAModelService * modelService;
+@property(nonatomic, strong, readonly) OBALocationManager * locationManager;
+
+/**
+ *  This block, if set, is called whenever refreshSettings also refreshes a region.
+ */
+@property (nonatomic, copy) dispatch_block_t regionRefreshed;
```

Comment:
> This blocks only purpose is to maintain legacy business logic compatibility

Here's my perspective on this: this code is still going to be here in three years. Some day, someone will come along and add code that overwrites that copied block. Maybe it'll slip through a code review, and then the app will unexpectedly stop working correctly in a way that will be extremely difficult to track down.

> that is really not the point here

The point for me is to ensure that whoever works on OneBusAway after us at some point in the future doesn't unexpectedly shoot themselves in the foot. Let's do the right thing not just for today but also for tomorrow.

> A delegate can be overwritten just as easily

True, so let's use notifications. Or, if you're dead-set on using blocks, change the `regionRefreshed` property to be a pair of methods called something like `-addRegionRefreshedHandler:` which will vend an opaque token of some kind that is later passed to `-removeRegionRefreshedHandler:` to remove that block.